lab.R

```diff
@@ -99,44 +105,64 @@ daily_diff = function(x, order_var) {
-#' 
+daily_diff = filtered_df %>% 
+    group_by(county, fips) %>% 
+    arrange(date) %>% 
+    mutate(cases =daily_diff(cases, date)) %>% 
+    mutate(deaths = daily_diff(deaths,date)) %>% 
+    ungroup()
+# There are NAs for 2020-08-02 because 2020-08-02 is the first day in the dataset, so lag function doesn't suit it. 
 
 #' 3. *Finally we need to calculate rates per 1 million residents.  Write a pipe that takes `daily_diff` as input, joins it with the `pop` dataframe using appropriate variables, removes any rows with missing FIPS codes, and constructs the variables `cases_per_pop` and `deaths_per_pop`.  When constructing these variables, multiply by `per_pop` to get rates per 1 million residents.  Assign the result to `covid_df`, since this contains the Covid data for our analysis.*  
-#' 
+covid_df = full_join(daily_diff, pop) %>% 
+    drop_na(fips) %>% 
+    mutate(cases_per_pop = (cases/population)*1000000) %>% 
+    mutate(deaths_per_pop = (deaths/population)*1000000)
+    
+
 
 
 
 #' # Problem 4 #
 #' 1. *To explore these time-series data visually, we'll want to use line plots of cases or deaths over time.  The line group needs the `group` aesthetic to determine which values should be treated as part of a single line.  Uncomment and fill in the blanks to plot cases per 1,000,000 residents over time for each county.*  
 #' 
 
-# ggplot(covid_df, aes(---, ---, group = ---)) +
-#     geom_line()
+ggplot(covid_df, aes(date,cases_per_pop , group = county)) +
+    geom_line()
 
 #' 2. *Because there are so many counties, the lines are heavily overplotted.  Modify your code from the last problem to facet by county.  Try both `scales = 'fixed'` and `scales = 'free_y'`.* 
-#' 
+plot_free = ggplot(covid_df, aes(date,cases_per_pop , group = county)) +
+    geom_line()+
+    facet_wrap(~county, scales = "free_y")
+
+plot_fixed = ggplot(covid_df, aes(date,cases_per_pop , group = county)) +
+    geom_line()+
+    facet_wrap(~county, scales = "fixed")
 
 #' 3. *The plot indicates that, on a few days, some counties gain or lose thousands of cases per million residents.  What's up this that?  Hints:  `plotly::ggplotly()` to create an interactive version of the most recent plot.  Use `filter()` to narrow down the data to particular counties during short periods of time, and `View()` to peruse the data after filtering.*  
-#' 
-#' 
-#' 
+plotly::ggplotly(plot_fixed)
+covid_df %>% filter(county == 'Lassen')
+covid_df %>% filter(county == 'Merced')
+covid_df %>% filter(county == 'Monterey')
+
+# I noticed that Lassen, Merced, and Monterey (and some other counties) gain or lose thousands of cases per million residents on a few days. It reflects the variaion of new confirmed cases. 
```

Comment:
Okay, but why do these counties see these massive swings?  Why not San Francisco, Sacramento, or LA?